code/game/machinery/stasis.dm

```diff
@@ -20,7 +21,12 @@
 
 // dir check for buckle_lying state
 /obj/machinery/stasis/Initialize()
-	RegisterSignal(src, COMSIG_ATOM_DIR_CHANGE, PROC_REF(dir_changed)) //This gets called later during initialization
+	RegisterSignal(src, COMSIG_ATOM_DIR_CHANGE, PROC_REF(dir_changed))
+	switch(dir)
```

Comment:
call dir_changed instead of copypasting.

Reply:
I thought of doing that, but wasn't sure at the time since I've gotten compiler warnings for mixing procs that don't sleep with ones that do.

Reply:
Neither proc sleeps

Reply:
In the case that one does and the other requires nonsleeping you can use INVOKE_ASYNC()